source/Detector/Amptek/AmptekSDD123Server.h

```diff
@@ -0,0 +1,209 @@
+	/// Sets a text configuration value. Must be the format of CMD=VALUE;
+	void textConfigurationSet(QString singleCommandRequest);
+
+	/// read the reply datagram
+	bool readReplyDatagram(int &id, QByteArray &datagram);
+
+	/// set the request interval timer
+	void setRequestIntervalTimer(int mSec);
+	/// set the spectrum packet receiver
+	void setSpectrumPacketReceiver(QObject *spectrumPacketReceiver);
+	/// set the confirmation packet receiver
+	void setConfirmationPacketReceiver(QObject *confirmationPacketReceiver);
+
+signals:
+	/// signal to acknowledge OK
+	void acknowledgeOK();
+	/// signal to indicate the spectrum data is ready
+	void spectrumDataReady(const QByteArray &spectrumData, int numChannels);
+	/// signal to indicate the status data is ready
+	void statusDataReady(const QByteArray &statusData);
+	/// signal to indicate all parameters text configuration is ready
+	void allParametersTextConfiguration(const QString &textConfiguration);
+	/// signal to indicate the response packet is ready
+	void responsePacketReady();
+
+	/// signal to indicate system detected drop packet
+	void dropPacketDetected();
+	/// signal to indicate system detected unrequested packet
+	void unrequestedPacketDetected();
+
+	/// signal to indicate a reply datagram is ready
+	void replyDatagram(int id, const QByteArray &datagram);
+
+	/// signal to indicate detector availability changed
+	void detectorUnavailableChanged(bool detectorUnavailable);
+	/// signal to indicate fatal timeout
+	void fatalTimeout();
+
+	/// signal to indicate that the server is going to change to configuration state
+	void serverAboutToChangeToConfigurationState();
+	/// signal to indicate that the server changed to configuration state
+	void serverChangedToConfigurationState();
+	/// signal to indicate that the server changed to dwell state
+	void serverChangedToDwellState();
+
+protected slots:
+	/// helper function to send request datagram
+	void sendRequestDatagram(AmptekSDD123Packet packet, int overrideTimeout = -1);
+	/// helper function to send request datagram synchronizedly
+	void sendSyncDatagram(AmptekSDD123Packet packet, int overrideTimeout = -1);
+	/// helper function to send fake request datagram
+	void fakeSendDatagram(AmptekSDD123Packet packet, int overrideTimeout = -1);
+
+	/// helper function to percess pending diagrams
+	void processPendingDatagrams();
+	/// helper function to handle reponse packet ready signal
+	void onResponsePacketReady();
+
+	/// helper function to synchonize request
+	void syncRequest(int timeout = -1);
+	/// helper function to fake synchonize request
+	void fakeSyncRequest(int timeout = -1);
+	/// helper function to handle request datagram timeout
+	void requestDatagramTimeout();
+	/// helper function to handle request sync datagram timeout
+	void syncDatagramTimeout();
+
+protected:
+	/// helper function to handle the local timedout message queue for the responsePacket
+	void processLocalStoredPacket(AmptekSDD123Packet responsePacket);
```

Comment:
These ones in particular are probably very important to have the `const XYZ &` arguments.  Especially the `QByteArray`'s.